### PR TITLE
Example for Kafka TLS Authentication

### PR DESCRIPTION
**Pull request by itsmurugappan**
## Proposed Changes 

- Add a an example to show KafkaSource TLS authentication
- Correct some formatting issues.

### CODE REVIEW

File: docs/eventing/samples/kafka/source/README.md [hunk 1]
```diff
@@ -247,23 +245,76 @@ You can specify the key deserializer among four types:
...
+    sink:
+      ref:
+        apiVersion: serving.knative.dev/v1
+        kind: Service
+        name: event-display
+   ```
+
+## Connecting to a TLS enabled Kafka broker
+
+The KafkaSource supports TLS and SASL authentication methods. For enabling TLS authentication, please have the below files
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:269]: Remove trailing whitespace:
```suggestion
The KafkaSource supports TLS and SASL authentication methods. For enabling TLS authentication, please have the below files
```

File: docs/eventing/samples/kafka/source/README.md [hunk 2]
```diff
@@ -247,23 +245,76 @@ You can specify the key deserializer among four types:
...
+   ```
+
+   $ kubectl create secret generic cacert --from-file=caroot.pem
+   secret/cacert created
+
+   $ kubectl create secret generic key --from-file=key.pem
+   secret/key created
+
+   $ kubectl create secret generic cert --from-file=certificate.pem
+   secret/cert created
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:286]: Remove trailing whitespace:
```suggestion
   secret/cert created
```

File: docs/eventing/samples/kafka/source/README.md [hunk 3]
```diff
@@ -247,23 +245,76 @@ You can specify the key deserializer among four types:
...
+            name: cacert
+   consumerGroup: knative-group
+   bootstrapServers: my-cluster-kafka-bootstrap.kafka:9092
+   topics: knative-demo-topic
+   sink:
+     ref:
+       apiVersion: serving.knative.dev/v1
+       kind: Service
+       name: event-display
+   ```
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:320]: Add trailing newline:
```suggestion
   ```

```

File: docs/eventing/samples/kafka/source/README.md [hunk 4]
```diff
@@ -247,23 +245,74 @@ You can specify the key deserializer among four types:
...
+    sink:
+      ref:
+        apiVersion: serving.knative.dev/v1
+        kind: Service
+        name: event-display
+   ```
+
+## Connecting to a TLS enabled Kafka broker
+
+The KafkaSource supports TLS and SASL authentication methods. For enabling TLS authentication, please have the below files
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:269]: Remove trailing whitespace:
```suggestion
The KafkaSource supports TLS and SASL authentication methods. For enabling TLS authentication, please have the below files
```

File: docs/eventing/samples/kafka/source/README.md [hunk 5]
```diff
@@ -247,23 +245,74 @@ You can specify the key deserializer among four types:
...
+            name: cacert
+   consumerGroup: knative-group
+   bootstrapServers: my-cluster-kafka-bootstrap.kafka:9092
+   topics: knative-demo-topic
+   sink:
+     ref:
+       apiVersion: serving.knative.dev/v1
+       kind: Service
+       name: event-display
+   ```
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:318]: Add trailing newline:
```suggestion
   ```

```

File: docs/eventing/samples/kafka/source/README.md (hunk 4 above)
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:269]: Remove trailing whitespace:
```suggestion
The KafkaSource supports TLS and SASL authentication methods. For enabling TLS authentication, please have the below files
```

File: docs/eventing/samples/kafka/source/README.md [hunk 6]
```diff
@@ -247,23 +245,74 @@ You can specify the key deserializer among four types:
...
+            name: cacert
+   consumerGroup: knative-group
+   bootstrapServers: my-secure-kafka-bootstrap.kafka:443
+   topics: knative-demo-topic
+   sink:
+     ref:
+       apiVersion: serving.knative.dev/v1
+       kind: Service
+       name: event-display
+   ```
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:318]: Add trailing newline:
```suggestion
   ```

```

File: docs/eventing/samples/kafka/source/README.md [hunk 7]
```diff
@@ -247,23 +245,75 @@ You can specify the key deserializer among four types:
...
+
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:319]: Remove trailing whitespace:
```suggestion

```

Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:319]: Add trailing newline:
```suggestion
   

```

File: docs/eventing/samples/kafka/source/README.md [hunk 8]
```diff
@@ -247,23 +245,75 @@ You can specify the key deserializer among four types:
...
+## Connecting to a TLS enabled Kafka broker
+
+The KafkaSource supports TLS and SASL authentication methods. For enabling TLS authentication, please have the below files
+
+* CA Certificate
+* Client Certificate and Key
+
+KafkaSource expects these files to be in pem format, if it is in other format like jks , please convert to pem.
```
Comment (mpetason) [inline docs/eventing/samples/kafka/source/README.md:274]:
```suggestion
KafkaSource expects these files to be in pem format, if it is in other format like jks, please convert to pem.
```

File: docs/eventing/samples/kafka/source/README.md [hunk 9]
```diff
@@ -124,8 +124,10 @@ You must ensure that you meet the [prerequisites listed in the Apache Kafka over
      name: kafka-source
    spec:
      consumerGroup: knative-group
-     bootstrapServers: my-cluster-kafka-bootstrap.kafka:9092 #note the kafka namespace
-     topics: knative-demo-topic
+     bootstrapServers:
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:127]: Remove trailing whitespace:
```suggestion
     bootstrapServers:
```

File: docs/eventing/samples/kafka/source/README.md [hunk 10]
```diff
@@ -124,8 +124,10 @@ You must ensure that you meet the [prerequisites listed in the Apache Kafka over
      name: kafka-source
    spec:
      consumerGroup: knative-group
-     bootstrapServers: my-cluster-kafka-bootstrap.kafka:9092 #note the kafka namespace
-     topics: knative-demo-topic
+     bootstrapServers: 
+     - my-cluster-kafka-bootstrap.kafka:9092 # note the kafka namespace
+     topics:
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:129]: Remove trailing whitespace:
```suggestion
     topics:
```

File: docs/eventing/samples/kafka/source/README.md [hunk 11]
```diff
@@ -247,23 +247,79 @@ You can specify the key deserializer among four types:
...
+   ```yaml
+   apiVersion: sources.knative.dev/v1alpha1
+   kind: KafkaSource
+   metadata:
+    name: kafka-source
+    labels:
+      kafkasources.sources.knative.dev/key-type: int
+   spec:
+    consumerGroup: knative-group
+    bootstrapServers:
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:260]: Remove trailing whitespace:
```suggestion
    bootstrapServers:
```

File: docs/eventing/samples/kafka/source/README.md [hunk 12]
```diff
@@ -247,23 +247,79 @@ You can specify the key deserializer among four types:
...
+   kind: KafkaSource
+   metadata:
+    name: kafka-source
+    labels:
+      kafkasources.sources.knative.dev/key-type: int
+   spec:
+    consumerGroup: knative-group
+    bootstrapServers: 
+    - my-cluster-kafka-bootstrap.kafka:9092 # note the kafka namespace
+    topics:
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:262]: Remove trailing whitespace:
```suggestion
    topics:
```

File: docs/eventing/samples/kafka/source/README.md [hunk 13]
```diff
@@ -247,23 +247,79 @@ You can specify the key deserializer among four types:
...
+        key:
+          secretKeyRef:
+            key: tls.key
+            name: kafka-secret
+        caCert:
+          secretKeyRef:
+            key: caroot.pem
+            name: cacert
+   consumerGroup: knative-group
+   bootstrapServers:
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:315]: Remove trailing whitespace:
```suggestion
   bootstrapServers:
```

File: docs/eventing/samples/kafka/source/README.md [hunk 14]
```diff
@@ -247,23 +247,79 @@ You can specify the key deserializer among four types:
...
+            key: tls.key
+            name: kafka-secret
+        caCert:
+          secretKeyRef:
+            key: caroot.pem
+            name: cacert
+   consumerGroup: knative-group
+   bootstrapServers: 
+   - my-secure-kafka-bootstrap.kafka:443
+   topics:
```
Comment (mattmoor) [inline docs/eventing/samples/kafka/source/README.md:317]: Remove trailing whitespace:
```suggestion
   topics:
```

**Pull request merged**